apps/health_campaign_field_worker_app/lib/data/repositories/local/household.dart

```diff
@@ -0,0 +1,171 @@
+import '../../data_repository.dart';
+import '../../local_store/sql_store/sql_store.dart';
+
+class HouseholdLocalRepository
+    extends LocalRepository<HouseholdModel, HouseholdSearchModel> {
+  HouseholdLocalRepository(super.sql, super.opLogManager);
+
+  @override
+  FutureOr<List<HouseholdModel>> search(HouseholdSearchModel query) async {
+    final selectQuery = sql.select(sql.household).join(
+      [
+        leftOuterJoin(
+          sql.householdAddress,
+          sql.householdAddress.address.equalsExp(
+            sql.household.clientReferenceId,
+          ),
+        ),
+        leftOuterJoin(
+          sql.address,
+          sql.address.clientReferenceId.equalsExp(
+            sql.householdAddress.address,
+          ),
+        ),
+      ],
+    );
+    final results = await (selectQuery
+          ..where(
+            buildAnd(
+              [
+                if (query.clientReferenceId != null)
+                  sql.household.clientReferenceId.equals(
+                    query.clientReferenceId,
+                  ),
+                if (query.id != null)
+                  sql.household.id.equals(
+                    query.id,
+                  ),
+                if (query.tenantId != null)
+                  sql.household.tenantId.equals(
+                    query.tenantId,
+                  ),
+              ],
+            ),
+          ))
+        .get();
+
+    return results.map((e) {
+      final household = e.readTable(sql.household);
+      final address = e.readTable(sql.address);
+
+      return HouseholdModel(
+        id: household.id,
+        tenantId: household.tenantId,
+        clientReferenceId: household.clientReferenceId,
+        memberCount: household.memberCount,
+        rowVersion: household.rowVersion,
+        address: AddressModel(
+          tenantId: address.tenantId,
+          clientReferenceId: address.clientReferenceId,
+          doorNo: address.doorNo,
+          latitude: address.latitude,
+          longitude: address.longitude,
+          locationAccuracy: address.locationAccuracy,
+          addressLine1: address.addressLine1,
+          addressLine2: address.addressLine2,
+          city: address.city,
+          pincode: address.pincode,
+          locality: BoundaryModel(
+            clientReferenceId: '',
+            code: '',
+            name: '',
+          ),
+          type: address.type,
+        ),
+      );
+    }).toList();
+  }
+
+  @override
+  FutureOr<void> create(HouseholdModel entity) async {
+    final addressValue = entity.address;
+
+    final householdCompanion = _getHouseholdCompanion(entity);
+    final addressCompanion = _getAddressCompanion(addressValue);
+
+    await sql.batch((batch) async {
+      batch.insert(sql.household, householdCompanion);
+      batch.insert(sql.address, addressCompanion);
+
+      batch.insert(
+        sql.householdAddress,
+        HouseholdAddressCompanion.insert(
+          clientReferenceId: const Uuid().v1(),
```

Comment:
let's use uuid v4 everywhere since v4 is random and v1 is time based. Also, since we're generating a lot of ids on the client, should we centralize generation via a utility method so any future changes to uuid generation are a one-place change?

Reply:
deferring fix to a future PR